classes/class-paysoncheckout-for-woocommerce-confirmation.php

```diff
@@ -107,12 +107,14 @@ public function confirm_recurring_payson_order( $order_id ) {
 		$subscription_id = ( null !== WC()->session && ! empty( WC()->session->get( 'payson_payment_id' ) ) ) ? WC()->session->get( 'payson_payment_id' ) : get_post_meta( $order_id, '_payson_checkout_id', true );
 		$subcriptions    = wcs_get_subscriptions_for_order( $order_id );
 		foreach ( $subcriptions as $subcription ) {
-			update_post_meta( $subcription->get_id(), '_payson_subscription_id', $subscription_id );
```

Comment:
The meta data is saved to the subscription (not the order) so you should update the subscription instead of the order.

```
$subscription->update_meta_data( '_payson_subscription_id', $subscription_id );
$subscription->save();
```